flux2/templates/helm-controller.yaml

```diff
@@ -13,6 +13,9 @@ spec:
   template:
     metadata:
       annotations:
+        {{- if .Values.podAnnotations }}
```

Comment:
same here, you might have a problem if .Values.podAnnotations will include the prometheus annotations, maybe consider to move the prometheus to the code (getFluxValues) and do the override merge there?